engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextCompositor.java

```diff
@@ -157,8 +159,9 @@ private TextArea getNextTextArea(int maxLineWidth) {
 			textArea.setMaxWidth(maxLineWidth);
 			textArea.setWidth(0);
 			addWordIntoTextArea(textArea, remainWord);
+			textArea.setKeepLastSHY(remainWord.isKeepLastSHY());
 			remainWord = null;
-			return textArea;
+			return textArea; // Why do we return here already?
```

Comment:
Why add your confusion to the source code?

Reply:
I think this return statement is a weakness in the algorithm used.
It results in the next line consisting of *two* TextAreas.
Say you have the word "super-cali-fragi-listic-expli-ali-docious" (I'm using minus signs instead of SHY here for clarity).
Assume this word appears at the end of line 1, and only part of it fits into line 1.
So line 1 consists of one TextArea ["..... This is an super-"]
Line 2 consists of two TextAreas: ["cali-", "fragilisticexplialidocious idea"].
For PDF rendering, we have to remove the trailing "-" from the first TextArea, which was quite easy to achieve.

But this also results in *two* text runs in the PDF for line 2 (didn't verify this).
So in the resulting PDF line 2 is logically ["cali", "fragilisticexplialidocious idea"].

This probably makes it a little harder for accessibility software to understand that in fact this should read
"califragilisticexplialidocious idea".

Also the PDF is a few bytes longer than needed.

The rest of the code tries to stuff as many "words" as possible into one TextArea, but in this place, it results in an additional TextArea containing only one "word" ("cali-" in my example).

So this return statement kind of contradicts the idea of the remaining code.

I didn't dare to change it, though, because I needed to focus on the task at hand.

Reply:
Yes, that is a great analysis which I would not have guessed from `// Why do we return here already?`